netZooPy/panda/panda.py

```diff
@@ -405,30 +414,121 @@ def panda_loop(self, correlation_matrix, motif_matrix, ppi_matrix):
         def t_function(x, y=None):
             '''T function.'''
             if y is None:
-                a_matrix = np.dot(x, x.T)
-                s = np.square(x).sum(axis=1)
-                a_matrix /= np.sqrt(s + s.reshape(-1, 1) - np.abs(a_matrix))
+                a_matrix = xp.dot(x, x.T)
+                s = xp.square(x).sum(axis=1)
+                a_matrix /= xp.sqrt(s + s.reshape(-1, 1) - xp.abs(a_matrix))
+            else:
+                a_matrix = xp.dot(x, y)
+                a_matrix /= xp.sqrt(xp.square(y).sum(axis=0) + xp.square(x).sum(axis=1).reshape(-1, 1) - xp.abs(a_matrix))
+            return a_matrix
+
+        def update_diagonal(diagonal_matrix, num, alpha, step):
+            '''Update diagonal.'''
+            xp.fill_diagonal(diagonal_matrix, xp.nan)
+            diagonal_std = xp.nanstd(diagonal_matrix, 1)
+            diagonal_fill = diagonal_std * num * math.exp(2 * alpha * step)
+            xp.fill_diagonal(diagonal_matrix, diagonal_fill)
+
+        def gt_function(x, y=None):
```

Comment:
I think you created `gt_function` as gpu version of `t_function`, in this case we probably should be the `t_function` as in the original version

Reply:
this xp is what i was trying to get at-- and explains your last comment on why there is another tfunction -- cupy has the capability to notice IF a gpu is present and then set xp to either np or cp -- making the code agnostic. see the bottom of this page. i tried but couldnt get colab to recognize the gpu automatically via xp (ie xp would = cp) even tho cp commands would work when explicitly made. So I figured I would PR the state it is in (requiring flag) to show you and then tinker with it in future remove the flag once xp = cp.get_array_module(2) sets xp to "cp" and not "np"